aggregators/mtg_cards/cards.js

```diff
@@ -1,87 +1,79 @@
 const request = require("request");
+const _ = require("lodash");
+
 class MtgCardLoader {
     constructor() {
         this.cardApi = "https://api.magicthegathering.io/v1/cards?name=";
     }
+    cardToString(card) {
+        const manaCost = card.manaCost ? " " + card.manaCost : "";
+        const cardInfo = ["**" + card.name + "**" + manaCost];
+        if (card.type) {
+            cardInfo.push(card.type);
+        }
+        if (card.text) {
+            cardInfo.push(card.text.replace(/\*/g, '\\*'));
+        }
+        if (card.loyalty) {
+            cardInfo.push(card.loyalty);
+        }
+        if (card.power) {
+            cardInfo.push(card.power.replace(/\*/g, '\\*') + "/" + card.toughness.replace(/\*/g, '\\*'));
+        }
+        if (card.printings) {
+            cardInfo.push(card.printings.join(", "));
+        }
+        cardInfo.push("http://magiccards.info/query?q=!" + encodeURIComponent(card.name));
+        return cardInfo.join("\n");
+    }
+    findCard(cardName, cards) {
+        // create an array containing each card exactly once, preferring cards with image
+        const [cardsWithImage, cardsWithoutImage] = _.partition(cards, "imageUrl");
+        const differentCardsWithoutImage = _.differenceBy(cardsWithoutImage, cardsWithImage, "name");
+        const uniqCards = _.uniqBy(_.concat(cardsWithImage, differentCardsWithoutImage), "name");
 
-    find(cardName, callback) {
+        // Look for an exact match by name
+        let card = uniqCards.find(c => c.name.toLowerCase() === cardName);
+        if (!card){
+            // Use a hit which shares the longest common prefix with the search term
+            card = _.maxBy(uniqCards, c => longestCommonPrefix(cardName, c.name.toLowerCase()));
```

Comment:
Maybe I've been overthinking this. You're right that there probably aren't any search results where we only have a partial match on the prefix and still want to prefer that result.